pandas/tests/frame/test_dtypes.py

```diff
@@ -669,6 +670,17 @@ def test_arg_for_errors_in_astype(self):
 
         df.astype(np.int8, errors='ignore')
 
+    @pytest.mark.parametrize("num", [1.0, 1])
+    @pytest.mark.parametrize("struct", [pd.Series, pd.DataFrame])
+    @pytest.mark.parametrize('op', [ops.add, ops.mul, ops.floordiv, ops.sub])
+    def test_assert_list_and_bool_coerce(self, num, struct, op):
+        # issue 18549
+        target_type = np.array([op(num, num)]).dtype
+        res = op(struct([True]), num).dtypes
```

Comment:
actually compare the result and directly construct the expected value; use ``assert_frame_equal``